### PR TITLE
416 425 merging

### PR DESCRIPTION
**Pull request by jum-s**
This PR integrates the last features required by 416 user api and 425 oauth
It opens endpoint with valid authentication process.
As a third party user, i can now access my own information through `/user` endpoint which fetch user store.
It also improve several api service:
- obsolete testing routes are deleted (`/users/:id`, `/secretPage`)
- set cookie is secured
- user store views byId update

### CODE REVIEW

**Comment by jum-s** [comment]
should be good to go

**Comment by almereyda** [comment]
Sorry, I was sick last week.
After deploying this branch to `staging`, a private browser window is immediately presented with a `connect.sid` cookie for the whole `.allmende.io` domain before attemping to call `/auth`. Also the server log contains a warning:

```
2017-11-11T00:53:43.444683132Z app[web.1]: -----
2017-11-11T00:53:45.551343979Z app[web.1]: Warning: connect.session() MemoryStore is not
2017-11-11T00:53:45.551373090Z app[web.1]: designed for a production environment, as it will leak
2017-11-11T00:53:45.551376178Z app[web.1]: memory, and will not scale past a single process.
```

**Comment by almereyda** [comment]
The use of a cookie name of `cookie.sid` and a domain scope of `allmende.io` for it breaks again the login at https://hack.allmende.io due to a name and scope collision.

**Comment by jum-s** [comment]
I dont think refusing this PR the best, nothing is wrong here, the branch works.. not perfectly but it does. To answer your questions: 

- the cookie name is just a default, doesnt change anything in the behavior.
- The cross domain scope has been deleted in the commit `7327bb3 `, are you talking about the cookie path ? that should be for `/data` ?
- According to https://stackoverflow.com/questions/10760620/using-memorystore-in-production' i dont think storing cookies in a memorystore (aka couchdb) is necessary since the **warning** (only) about memory leaks is about storing expired cookies of users that dont come back to the platform, which is an acceptable risk for the moment.

File: server/middlewares.js [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
   common: [
     require('body-parser').json(),
     require('method-override')(),
     require('errorhandler')({ dumpExceptions: true, showStack: true }),
-    cors
+    cors,
+    passport.initialize,
+    passport.session({
+      cookie: {
+        key: CONFIG.get('store.host'),
```
Comment (almereyda) [inline server/middlewares.js]: This is the hostname of the CouchDB instance. Why is it added or serialised into the Passport cookie?

File: server/middlewares/passport.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    done(err)
+  })
+})
+
+// assign to passport current session a default strategy
+passport.use('gitlab', gitlabStrategy.auth)
+
+const redirectionParams = {
+  failureRedirect: '/',
+  successReturnToOrRedirect: CONFIG.get('auth.clientEndpoint'),
```
Comment (almereyda) [inline server/middlewares/passport.js]: If we want generalised clients also other than the current editor, we may want to send the users back to where they came from.

File: server/middlewares/passport.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+const authenticate = passport.authenticate('gitlab', redirectionParams)
+
+function ensureAuthenticated (req, res, next) {
+  //technical debt here : req == Id provider request (not client reques), returnTo will be triggered in the redirectionParams, and will redirect to the provider page, which is not the oauth conventionnal behavior.
+  //better behavior would be to returnTo client origin, but it cannot be set here. Possible implementation : client request send a query parameter containing url origin of the client, which can be bind to the session before ensureAuthenticated is triggered
+  if (req.session.passport && _.isUuid(req.session.passport.user)) {
+    next()
+  } else {
+    req.session.returnTo = CONFIG.get('auth.clientEndpoint')
```
Comment (almereyda) [inline server/middlewares/passport.js]: Same as above.

File: server/middlewares/passport.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+const CONFIG = require('config')
+const __ = CONFIG.universalPath
+const _ = __.require('lib', 'utils')
+const session = require('express-session')
+const passport = require('passport')
+const User = __.require('controllers', '/users/lib/users')
+const gitlabStrategy = require('./strategies/gitlab')
```
Comment (almereyda) [inline server/middlewares/passport.js:7]: Was there any specific reason why we're not using https://github.com/fh1ch/passport-gitlab2/blob/master/lib/strategy.js?

**Review by almereyda — APPROVED** [review]
This pull request has been tested and works partially according to the specification.

**Pull request merged**